src/go/k8s/api/vectorized/v1alpha1/cluster_types.go

```diff
@@ -183,6 +183,37 @@ type ClusterSpec struct {
 
 	// The name of the ServiceAccount to be used by the Redpanda pods
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+
+	// NodePools runs a StatefulSet for every NodePool. All these produce pods,
+	// that join the cluster, but are steered independently (especially replica
+	// count).
+	NodePools []*NodePoolSpec `json:"nodePools,omitempty"`
+}
+
+// NodePoolSpec defines a NodePool. NodePools have their own:
+// NodeSelector, so they can be scheduled on specific cloud provider Node Pools.
+// Storage, as different NodePools may have different disk shapes.
+// Replicas, so they can be controlled independently
+// Resources, because this is tied strongly to the actual machine shape backing the NodePool.
+type NodePoolSpec struct {
+	Name string `json:"name"`
+	// Replicas determine how big the node pool will be.
+	// +kubebuilder:validation:Minimum=0
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
I would suggest to extend it as @nicolaferraro did in https://github.com/redpanda-data/redpanda/commit/27119769e88dbb840564284f1f085bd93437f61f 

to have:
* Replicas - A copy of Spec.Replicas with generation that was being reconciled
* CurrentReplicas - What is actually set in StatefulSet in current generation 
* ReadyReplicas - How many Redpandas are actually in ready state. It would flip back and fourth as Readiness probe is tided to cluster wide cluster health

Reply:
My bad you are right.